apps/web/components/ui/write-review-form.tsx

```diff
@@ -50,23 +50,76 @@ export default function WriteReviewForm({ onSubmit, previousReview }: WriteRevie
 		content: previousReview?.description || ""
 	});
 
+	useEffect(() => {
+		if (!previousReview) {
+			const savedAcademicYear = localStorage.getItem("reviewFormAcademicYear");
+			const savedRating = localStorage.getItem("reviewFormRating");
+			const savedDescription = localStorage.getItem("reviewFormDescription");
+
+			if (savedAcademicYear) setAcademicYear(savedAcademicYear);
+			if (savedRating) setRating(parseFloat(savedRating));
+			if (savedDescription && markdownEditor) markdownEditor.commands.setContent(savedDescription);
+		}
+	}, [previousReview, markdownEditor]);
+
+	//to save into localstorage
+	useEffect(() => {
+		if (!previousReview) {
+			const savedAcademicYear = localStorage.getItem("reviewFormAcademicYear");
+			const savedRating = localStorage.getItem("reviewFormRating");
+			const savedDescription = localStorage.getItem("reviewFormDescription");
+
+			if (savedAcademicYear) setAcademicYear(savedAcademicYear);
+			if (savedRating) setRating(parseFloat(savedRating));
+			if (savedDescription && markdownEditor) markdownEditor.commands.setContent(savedDescription);
+		}
+	}, [previousReview, markdownEditor]);
+
```

Comment:
what's the difference between these two?

Reply:
Also if I understand correctly, the reason you do
```js
if (!previousReview) {
...
}
```
is to handle the case where user edits their reviews right?

Reply:
yess yess, that's a duplication